### DISCUSSION TITLE
What version number should builds of the main branch have?

### DISCUSSION THREAD

**Discussion by mtfriesen** [microsoft/ebpf-for-windows#2346] [Q&A]
eBPF has released version `0.8.0` and the XDP project would like to consume fixes from newer builds of the `main` branch. However, `main` is still producing `0.7.0` artifacts - is this intended? Should `main` always have version number equal or greater than of any release?

**Comment by ballyalley-o** [comment]
No, it is not intended for the main branch of eBPF to produce 0.7.0 artifacts when the released version is 0.8.0. The main branch should always have a version number equal or greater than any release. This is because the main branch is the unstable development branch of eBPF, and it should always contain the latest features and fixes. If the main branch is producing 0.7.0 artifacts, it means that there are new features and fixes that are not yet available in the released version. This can be a problem for projects like XDP that want to consume the latest fixes from the main branch.

In this case, it is likely that the maintainers of the eBPF project are still working on the 0.8.0 release and have not yet pushed the latest artifacts to the main branch. Once the 0.8.0 release is finalized, the main branch should be updated to 0.8.0 and it should start producing 0.8.0 artifacts.

In the future, it would be better if the eBPF project made it clear that the main branch should always have a version number equal or greater than any release. This would help projects like XDP to know when they can expect to consume the latest fixes from the main branch.

**Comment by gtrevi** [comment]
The release process was fixed from 0.9.0, so now `main` will always have the source code versioning up to date with the latest version.
